lib/proxy/router.go

```diff
@@ -181,19 +194,28 @@ func NewRouter(cfg RouterConfig) (*Router, error) {
 	return &Router{
 		clusterName:    cfg.ClusterName,
 		log:            cfg.Log,
+		emitter:        cfg.Emitter,
 		clusterGetter:  cfg.RemoteClusterGetter,
 		localSite:      localSite,
 		siteGetter:     cfg.SiteGetter,
 		tracer:         cfg.TracerProvider.Tracer("Router"),
 		serverResolver: cfg.serverResolver,
 	}, nil
+}
 
+// IdentityInfo is the information needed about a Teleport user's
+// identity to make an RBAC check.
+type IdentityInfo struct {
+	TeleportUser  string
+	Login         string
+	Certificate   *ssh.Certificate
+	AccessChecker services.AccessChecker
```

Comment:
Why is the access checker a part of the identity? What is the relation between them? Can you explain that in a comment or change it?